module/zfs/arc.c

```diff
@@ -7647,6 +7647,8 @@ arc_init(void)
 	if (arc_c < arc_c_min)
 		arc_c = arc_c_min;
 
+	arc_register_hotplug();
```

Comment:
There needs to be a matching `arc_unregister_hotplug()` wrapper which calls `unregister_memory_notifier()`.  Otherwise we can get in to a situation where the modules are unloaded by the notifier remains registered but with an invalid memory address for where `arc_register_hotplug()` used to reside.